sockeye/inference.py

```diff
@@ -630,6 +635,62 @@ def make_input(sentence_id: int, sentence: str) -> TranslatorInput:
         tokens = list(data_io.get_tokens(sentence))
         return TranslatorInput(id=sentence_id, sentence=sentence.rstrip(), tokens=tokens)
 
+    def batch_translate(self, trans_inputs: List[TranslatorInput]) -> List[TranslatorOutput]:
+        """
+        Batch-translates a list of TranslatorInput, returns a list of TranslatorOutputs.
+        Splits oversized sentneces to sentence chunks of size less than max_input_length.
+
+        :param trans_input: List of TranslatorInputs as returned by make_input().
+        :return: List of translation results.
+        """
+        # TODO handle empty inputs, preferably outside inference.py?
+        # TODO avoid accepting and creating lists, use generators?
+
+        # keep input ids to be able to concatenate translated sentence chunks back
+        chunk_markers = []
+        input_chunks = []
+
+        # prepare chunks, keeping track of splitted inputs
+        for trans_input in trans_inputs:
+            if len(trans_input.tokens) > self.max_input_length:
+                logger.debug("Input %d has length (%d) that exceeds max input length (%d). Splitting into chunks of size %d.",
+                             trans_input.id, len(trans_input.tokens), self.buckets_source[-1], self.max_input_length)
+                token_chunks = list(utils.chunks(trans_input.tokens, self.max_input_length))
+                input_chunks.extend(token_chunks)
+                chunk_markers.extend([trans_input.id] * len(token_chunks))
+            else:
+                input_chunks.append(trans_input.tokens)
+                chunk_markers.append(trans_input.id)
+
+        # start translation in batches
+        translations = []  # type: List[Translation]
+        for i in range(0, len(input_chunks), self.batch_size):
+            logger.debug("Translating batch %d", i)
+            batch = input_chunks[i:i + self.batch_size]
+            # underfilled batch will be filled to a full batch size with empty inputs
```

Comment:
empty or just the first element of the batch?

Reply:
Right, the 1st